module/util/log.go

```diff
@@ -4,18 +4,24 @@ import (
 	"github.com/rs/zerolog"
 )
 
+// LogProgress takes a total and return function such that when called with a 0-based index
+// it prints the progress from 0% to 100% to indicate the index from 0 to (total - 1) has been
+// processed.
+// useful to report the progress of processing the index from 0 to (total - 1)
 func LogProgress(msg string, total int, logger *zerolog.Logger) func(currentIndex int) {
-	logThreshold := float64(10)
+	lookup := make(map[int]int)
+	for i := 1; i <= 10; i++ { // 10% to 100%
+		lookup[total/10*i] = i * 10 // percentage
```

Comment:
I believe I've also commented on this in the past.  please use order multiple before divide, or use parentheses.  it makes more mental power to read `total/10*i` than `i*total/10`